packages/fast-components-styles-msft/src/button/index.ts

```diff
@@ -1,16 +1,192 @@
 import { IDesignSystem } from "../design-system";
-import { ComponentStyles } from "@microsoft/fast-jss-manager";
+import { ComponentStyles, ICSSRules } from "@microsoft/fast-jss-manager";
 import { IButtonClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { IMSFTButtonClassNameContract } from "@microsoft/fast-components-class-name-contracts-msft";
+import { toPx } from "@microsoft/fast-jss-utilities";
+import { applyType } from "../utilities/typography";
+import * as Chroma from "chroma-js";
+import { isNullOrUndefined } from "util";
 
-const styles: ComponentStyles<IButtonClassNameContract, IDesignSystem> = {
+function applyTransaprentBackplateStyles(): ICSSRules<IDesignSystem> {
+    return {
+        extend: "button",
+        backgroundColor: (): string => {
+            return "transparent";
+        },
+        "&:hover": {
+            backgroundColor: (): string => {
+                return "transparent";
+            }
+        },
+        color: (config: IDesignSystem): string => {
+            return config.brandColor;
+        },
+        "&:disabled span:before, &[aria-disabled] span:before": {
+            background: "transparent"
+        },
+        "&:focus span:before, &:active span:before, &:hover span:before": {
+            // TODO: Issue #309 https://github.com/Microsoft/fast-dna/issues/309
+            background: (config: IDesignSystem): string => {
+                return config.brandColor;
+            }
+        },
+        "&:disabled, &[aria-disabled]": {
+            "&:hover": {
+                background: (): string => {
+                    return "transparent";
+                }
+            }
+        }
+    };
+}
+
+function applyMixedColor(incomingProperty: string, value?: number, alpha?: number, accent?: boolean): ICSSRules<IDesignSystem> {
+
+    function applyColors(config: IDesignSystem): string {
+        if (alpha) {
+            if (accent && isNullOrUndefined(value)) {
+                return Chroma(config.brandColor).alpha(alpha).css();
+            } else {
+                return Chroma.mix(config.foregroundColor, config.backgroundColor, value).alpha(alpha).css();
+            }
+        } else {
+            return Chroma.mix(config.foregroundColor, config.backgroundColor, value).css();
+        }
+    }

```

Comment:
Is this similar to flatten overlay? Can we pull this out as a utility so that we can apply to other components.

Reply:
We can pull out or change after a discussion. This simply mixes colors, and per your earlier comment, I pulled the gray out of mixing the foreground and background.

Reply:
Note: If we pull applyColors (suggest a more specific name to what it's doing if we extrapolate it out) it should remove the cognitive complexity issues on code climate